File: chess_java/src/chess/validators/RookMoveValidator.java
package chess.validators;

import chess.Board;
import chess.Spot;
import chess.utils.MessageHandler;

public class RookMoveValidator extends PieceMoveValidator {

    @Override
    public boolean validate(Board board, Spot start, Spot end) {
        return (start.getX() == end.getX() && Math.abs(start.getY() - end.getY()) > 0) ||
                (start.getY() == end.getY() && Math.abs(start.getX() - end.getX()) > 0);
    }

    @Override
    protected String getNotValidMessage() {
        return MessageHandler.getHellNoMessage();
    }
}

File: chess_cpp/validators/RookMoveValidator.cpp
//
// Created by Hrach Khachatryan on 12/15/19.
//

#include "RookMoveValidator.h"
#include "../utils/MessageHandler.h"

bool RookMoveValidator::validate(Board *board, Spot *start, Spot *end) {
    return (start->getX() == end->getX() && abs(start->getY() - end->getY()) > 0) ||
           (start->getY() == end->getY() && abs(start->getX() - end->getX()) > 0);
}

std::string RookMoveValidator::getNotValidMessage() {
    return MessageHandler::getHellNoMessage();
}
